### PR TITLE
Backend Integration tests (backend-tests/run script) always prints "tests failed"

### PR DESCRIPTION
**Pull request by merlin-northern**
ChangeLog:title
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

File: backend-tests/run [hunk 1]
```diff
@@ -145,12 +147,17 @@ parse_args "$@"
...
-    tmppath=$(mktemp /tmp/acceptance.XXXXXX)
     echo "-- tests failed, dumping logs to $tmppath"
-    $COMPOSE_CMD logs > $tmppath
+else
+    echo "-- tests successful, dumping logs to $tmppath"
 fi
 
+$COMPOSE_CMD logs > $tmppath
+
+
```
Comment (lluiscampos) [inline backend-tests/run]: Why this change? Not reflected in the commit message

**Comment by lluiscampos** [comment]
> but `get_container_exit_code` returns `"1"` or other return code,

get_container_exit_code returns 0 if all tests passed. I actually think that docker attach returns the exact same exit code...

>  so after line 98 `$failed` always has value, and in effect on line [148](https://github.com/mendersoftware/integration/blob/master/backend-tests/run#L148) it is always prints that tests failed.

This is what I don't understand. If that is true, the backend integration test job in GitLab should have been failing all this time.

File: backend-tests/run [hunk 2]
```diff
@@ -94,8 +94,10 @@ run_tests() {
     fi
 
     docker attach $cid || failed=1
-
-    failed=$(get_container_exit_code $cid)
+    rc=$(get_container_exit_code $cid)
+    if [ "$failed" != "1" ] && [ "$rc" != "0" ]; then 
+        failed=$rc
+    fi
```
[the remark this answers is not in the document]
Comment (merlin-northern) [reply backend-tests/run]: > > but `get_container_exit_code` returns `"1"` or other return code,
> 
> get_container_exit_code returns 0 if all tests passed. I actually think that docker attach returns the exact same exit code...
> 
> > so after line 98 `$failed` always has value, and in effect on line [148](https://github.com/mendersoftware/integration/blob/master/backend-tests/run#L148) it is always prints that tests failed.
> 
> This is what I don't understand. If that is true, the backend integration test job in GitLab should have been failing all this time.

the `failed` is only used to print the message, and every test run prints that tests failed.

[the remark this answers is not in the document]
Comment (lluiscampos) [reply backend-tests/run]: Okey, I finally understand what is going on!

> the `failed` is only used to print the message, and every test run prints that tests failed.

No: the `failed` is used to exit the script, see last line. This is the part I was talking about.

But then it was tested to contain any value and print "failed". That is what we should change to test that is different than 0.

[the remark this answers is not in the document]
Comment (merlin-northern) [reply backend-tests/run]: > Okey, I finally understand what is going on!
> 
> > the `failed` is only used to print the message, and every test run prints that tests failed.
> 
> No: the `failed` is used to exit the script, see last line. This is the part I was talking about.

I meant it is used in the logic, later it is returned, yes, but untouched, just as it was.

> 
> But then it was tested to contain any value and print "failed". That is what we should change to test that is different than 0.

so, you approved?

[the remark this answers is not in the document]
Comment (lluiscampos) [reply backend-tests/run]: I don't like it. But I approved it 4 hours ago, yes

**Comment by merlin-northern** [comment]
> > @lluiscampos I hope [this](https://github.com/mendersoftware/integration/pull/728/commits/1f9eb3b63dd0fa1ed02c651c0b41c9ebcf10d6f4) fixes all we talked about.
> 
> 👍 ❤️

I love it!

> Just remember to squash before the merge.

right! done!

> And thinking about it again: you were right since the beginning, the tests always printed "failed" (as your PR title indicated), it was me who got confused as the exit of the script was actually correct. I can get a bit stubborn some times ;)

Lluìs: do *not* ever change.
